tst/regression/utils/compare_analytic.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+
+################################################################################
+#   Sample error functions
+#
+#   Each error function accepts the gold and test data in the format
+#   and returns a error metric
+################################################################################
+
+
+def l1_err(gold, test):
+    return np.sum(np.abs(gold - test))
+
+
+def l2_err(gold, test):
+    return np.linalg.norm(gold - test)
+
+
+def linf_err(gold, test):
+    return np.max(np.abs(gold - test))
+
+
+def l1_rel_err(gold, test):
+    mask = gold != 0
+    return np.sum(np.abs((gold[mask] - test[mask]) / gold[mask]))
+
+
+def l2_rel_err(gold, test):
+    mask = gold != 0
+    return np.linalg.norm((gold[mask] - test[mask]) / gold[mask])
+
+
+def linf_rel_err(gold, test):
+    mask = gold != 0
+    return np.max(np.abs((gold[mask] - test[mask]) / gold[mask]))
+
+
+################################################################################
+#   Compare_analytic
+#
+#   Compares data in filename to a dictionary of analytic functions for some
+#   components
+################################################################################
+
+
+def compare_analytic(
+    filename, analytic_components, err_func=l2_err, tol=1e-12, quiet=False
+):
+    """Compares data in filename to analytic gold data in analytic_components.
+
```

Comment:
I think it might make sense to put this file in: scripts/python/packages/parthenon_tools/parthenon_tools, that way it will be installed with the other tools. Is there a particular reason you placed it here?

Reply:
I see why it's in regression, but I agree. I think this is more broadly useful and I would like access to it "downstream and installed."

Reply:
I agree. This started before the other tools were put together as an installable package. `compare_analytic.py` belongs with the other tools.